### ISSUE TITLE
Support async function 

### ISSUE THREAD

**Issue by cncolder** [okv/east#20]
[utils.js#L3](https://github.com/okv/east/blob/53651708f8bfafa6edf37513ee381e5d49384d4c/lib/utils.js#L3) limit cannot use async function

```js
exports.migrate = async (client, done) => {}
```

```bash
> toString.call(async function(){})
'[object AsyncFunction]'
> toString.call(async ()=>{})
'[object AsyncFunction]'
```

I suggest put 'AsyncFunction' in utils array. Then update [migrator.js#L299-L301](https://github.com/okv/east/blob/53651708f8bfafa6edf37513ee381e5d49384d4c/lib/migrator.js#L299-L301) like this:

```js
else if (!(utils.isFunction(migration.migrate) && utils.isAsyncFunction(migration.migrate)))
```

**Comment by okv** [comment]
I'll try to take a look at that on next weekend. Thanks for pointing that out.

**Comment by okv** [comment]
Looks like I missed the point at first look. East is not aware about async/await functionality and it will not await migrate/rollback functions. I mean it's not only validation issue. Currently I don't have time to add such support.

**Comment by cncolder** [comment]
No problem. East aware `done` callback.

```js
exports.migrate = async (client, done) => {
  await doSth()
  await cleanSth()
  done()
}
```

Currently I write like this:

```js
const migrate = async client => {
  await doSth()
}

exports.migrate = (client, done) => {
  migrate
    .then(() => done())
    .catch(done)
}
```

**Comment by okv** [comment]
ok, done, published east@0.5.5
